lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js

```diff
@@ -339,4 +343,21 @@ describe('LMSFilePicker', () => {
     wrapper.update();
     assert.isFalse(wrapper.isEmptyRender());
   });
+
+  describe('when no files are provided', () => {
+    beforeEach(() => {
+      fakeApiCall = sinon.stub().resolves([]); // no files returned
+    });
+
+    it('renders no file message with a help link', async () => {
+      const wrapper = renderFilePicker();
+      await fakeApiCall.returnValues[0];
+      wrapper.update();
```

Comment:
This `await ...; wrapper.update();` pair of statements follows the example of other tests, so I think it's OK to use it for the moment. This pattern is a little brittle though in that it will break if `LMSFilePicker` ever does any other additional async steps after the file list API call returns before it re-renders. A better solution is to wait for the `FileList` to be rendered in a non-loading state using the `waitForElement` helper:

```diff
diff --git a/lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js b/lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js
index 55f65b72..4218e475 100644
--- a/lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js
+++ b/lms/static/scripts/frontend_apps/components/test/LMSFilePicker-test.js
@@ -7,6 +7,7 @@ import ErrorDisplay from '../ErrorDisplay';
 import FileList from '../FileList';
 import LMSFilePicker, { $imports } from '../LMSFilePicker';
 import mockImportedComponents from '../../../test-util/mock-imported-components';
+import { waitForElement } from '../../../test-util/wait';
 
 describe('LMSFilePicker', () => {
   // eslint-disable-next-line react/prop-types
@@ -351,9 +352,7 @@ describe('LMSFilePicker', () => {
 
     it('renders no file message with a help link', async () => {
       const wrapper = renderFilePicker();
-      await fakeApiCall.returnValues[0];
-      wrapper.update();
-      const fileList = wrapper.find('FileList');
+      const fileList = await waitForElement(wrapper, 'FileList');
       assert.equal(
         fileList.prop('noFilesMessage').props.href,
         'https://fake_help_link'
```

Reply:
I like waitForElement a lot better, much easier  as it takes care of those details which are hard to remember. I'll switch all the tests in this file which could use that call, perhaps in a second commit.